screening/types/screening_ugc_violation_param.go

```diff
@@ -0,0 +1,153 @@
+	}
+
+	err = suvp.Context.ExtractFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to extract ScreeningUGCViolationParam.Context. %s", err.Error())
+	}
+
+	err = suvp.ScreenshotDataID.ExtractFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to extract ScreeningUGCViolationParam.ScreenshotDataID. %s", err.Error())
+	}
+
+	return nil
+}
+
+// Copy returns a new copied instance of ScreeningUGCViolationParam
+func (suvp ScreeningUGCViolationParam) Copy() types.RVType {
+	copied := NewScreeniGCUgcViolationParam()
+
+	copied.StructureVersion = suvp.StructureVersion
+	copied.Category = suvp.Category
+	copied.Reason = suvp.Reason
+	copied.Context = suvp.Context
+	copied.ScreenshotDataID = suvp.ScreenshotDataID
+
+	return copied
+}
+
+// Equals checks if the given ScreeningUGCViolationParam contains the same data as the current ScreeningUGCViolationParam
+func (suvp ScreeningUGCViolationParam) Equals(o types.RVType) bool {
+	if _, ok := o.(ScreeningUGCViolationParam); !ok {
+		return false
+	}
+
+	other := o.(ScreeningUGCViolationParam)
+
+	if suvp.StructureVersion != other.StructureVersion {
+		return false
+	}
+
+	if suvp.Category != other.Category {
+		return false
+	}
+
+	if suvp.Reason != other.Reason {
+		return false
+	}
+
+	if !suvp.Context.Equals(other.Context) {
+		return false
+	}
+
+	return suvp.ScreenshotDataID == other.ScreenshotDataID
+}
+
+// CopyRef copies the current value of the ScreeningUGCViolationParam
+// and returns a pointer to the new copy
+func (suvp ScreeningUGCViolationParam) CopyRef() types.RVTypePtr {
+	copied := suvp
+	return &copied
+}
+
+// Deref takes a pointer to the ScreeningUGCViolationParam
+// and dereferences it to the raw value.
+// Only useful when working with an instance of RVTypePtr
+func (suvp *ScreeningUGCViolationParam) Deref() types.RVType {
+	return *suvp
+}
+
+// String returns the string representation of the ScreeningUGCViolationParam
+func (suvp ScreeningUGCViolationParam) String() string {
+	return suvp.FormatToString(0)
+}
+
+// FormatToString pretty-prints the ScreeningUGCViolationParam using the provided indentation level
+func (suvp ScreeningUGCViolationParam) FormatToString(indentationLevel int) string {
+	indentationValues := strings.Repeat("\t", indentationLevel+1)
+	indentationEnd := strings.Repeat("\t", indentationLevel)
+
+	var b strings.Builder
+
+	b.WriteString("ScreeningUGCViolationParam{\n")
+	b.WriteString(fmt.Sprintf("%sCategory: %d,\n", indentationValues, suvp.Category))
+	b.WriteString(fmt.Sprintf("%sReason: %s,\n", indentationValues, suvp.Reason))
+	b.WriteString(fmt.Sprintf("%sContext: %s,\n", indentationValues, suvp.Context))
+	b.WriteString(fmt.Sprintf("%sScreenshotDataID: %s\n", indentationValues, suvp.ScreenshotDataID))
+	b.WriteString(fmt.Sprintf("%s}", indentationEnd))
+
+	return b.String()
+}
+
+// NewScreeniGCUgcViolationParam returns a new ScreeningUGCViolationParam
+func NewScreeniGCUgcViolationParam() ScreeningUGCViolationParam {
```

Comment:
```suggestion
// NewScreeningUGCViolationParam returns a new ScreeningUGCViolationParam
func NewScreeningUGCViolationParam() ScreeningUGCViolationParam {
```

The name of the new function